subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -84,12 +84,21 @@ static int wifi_scan(uint32_t mgmt_request, struct net_if *iface,
 	const struct device *dev = net_if_get_device(iface);
 	struct net_wifi_mgmt_offload *off_api =
 		(struct net_wifi_mgmt_offload *) dev->api;
+	struct wifi_scan_params *params = data;
 
 	if (off_api == NULL || off_api->scan == NULL) {
 		return -ENOTSUP;
 	}
 
-	return off_api->scan(dev, scan_result_cb);
+	if (!data || len != sizeof(*params)) {
+		return -EINVAL;
```

Comment:
This breaks compatibility with current applications, as this is an optional param, IMHO we should just ignore.